connectivity/netsocket/include/netsocket/NetworkInterface.h

```diff
@@ -387,8 +388,30 @@ class NetworkInterface: public DNS {
      * of both leads to undefined behavior.
      *
      *  @param status_cb The callback for status changes.
+     *  @return NSAPI_ERROR_OK on success
+     *  @return NSAPI_ERROR_NO_MEMORY if the function fails to create a new entry.
      */
+    nsapi_error_t add_event_listener(mbed::Callback<void(nsapi_event_t, intptr_t)> status_cb);
+#else
+    /** Add event listener for interface.
+     *
+     * This API allows multiple callback to be registered for a single interface.
+     * When first called, internal list of event handlers are created and registered to
+     * interface through attach() API.
+     *
+     * Application may only use attach() or add_event_listener() interface. Mixing usage
+     * of both leads to undefined behavior.
+     *
+     * @warning This version of the function does not use the `std::nothrow` feature. Subsequently,
```

Comment:
To fix the spellchecker:

 add nothrow to mbed-os/tools/test/travis-ci/doxy-spellchecker/ignore.en.pws